gxexternalproviders/src/main/java/com/genexus/db/driver/ExternalProviderS3.java

```diff
@@ -145,18 +151,22 @@ public String upload(String externalFileName, InputStream input, boolean isPriva
             if (externalFileName.endsWith(".tmp")) {
                 metadata.setContentType("image/jpeg");
             }
-            ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(bytes);
-            PutObjectResult result = client.putObject(new PutObjectRequest(bucket, externalFileName, byteArrayInputStream, metadata).withCannedAcl(getUploadACL(isPrivate)));
-            return ((AmazonS3Client) client).getResourceUrl(bucket, externalFileName);
+            String upload = "";
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: When objects are uploaded to Amazon S3 by using streams (either through an AmazonS3 client or TransferManager), you might encounter network connectivity or timeout issues.

**Fix**: The most reliable way to avoid a ResetException is to provide data by using a File or FileInputStream, which the AWS SDK for Java can handle without being constrained by mark and reset limits. If the input stream isn&#39;t a FileInputStream but does support mark and reset, you can set the mark limit by using the [setReadLimit](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/RequestClientOptions.html#setReadLimit-int-) method of [RequestClientOptions.](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/RequestClientOptions.html) Its default value is 128 KB. Setting the read limit value to one byte greater than the size of stream will reliably avoid a ResetException.

[Learn more](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/best-practices.html)